### PR TITLE
♻️(scss) refactor scss code

### PR DESCRIPTION
**Pull request by igobranco**
Refactor frontend scss code. 
Add `_all.scss` on each folder so the `_main.scss` could be more compact.
Add `!default` to homegrown Richie shared scss settings/variables.

#1687

### CODE REVIEW

**Review by jbpenrath — COMMENTED** [review]
This modification introduces deep changes about the manner to override stylesheets for our site factory. You should add entry about that in the UPGRADE.md.

**Comment by igobranco** [comment]
> This modification introduces deep changes about the manner to override stylesheets for our site factory. You should add entry about that in the UPGRADE.md.

Done.

File: src/frontend/scss/_main.scss [hunk 1]
```diff
@@ -20,116 +20,21 @@
...
-@import './components/templates/courses/cms/category_list';
-@import './components/templates/courses/cms/category_detail';
-@import './components/templates/courses/cms/course_detail';
-@import './components/templates/courses/cms/person_list';
-@import './components/templates/courses/cms/person_detail';
-@import './components/templates/courses/cms/blogpost_list';
-@import './components/templates/courses/cms/blogpost_detail';
-@import './components/templates/courses/cms/program_list';
-@import './components/templates/courses/cms/program_detail';
+@import './components/all';
```
Comment (jbpenrath) [inline src/frontend/scss/_main.scss:37]: Not a fan to have this file at the root of js/components/ folder.

Furthermore, within scss/ folder, we have a components/ directory so, from my point of view, it seems logic that those styles are imported through components/_all.scss.

What do you think? Do you see drawbacks ?

**Comment by jbpenrath** [comment]
One last little topic that I have in mind : 
In order to homegeneize naming accross frontend. What do you think to rename `_all.scss` files into `_index.scss`?

File: src/frontend/scss/_main.scss (hunk 1 above)
Comment (igobranco) [reply src/frontend/scss/_main.scss]: Ok. Done.

**Review by jbpenrath — APPROVED** [review]
Many thanks @igobranco 🚀 
It will grealty ease our upgrades through our site-factory ! 🤩

**Pull request merged**